api/src/main/java/org/eclipse/microprofile/reactive/messaging/Message.java

```diff
@@ -297,6 +449,32 @@ default Metadata getMetadata() {
         }
     }
 
+    /**
+     * Acknowledge negatively this message.
+     * <code>nack</code> is used to indicate that the processing of a message failed. The reason is passed as parameter.
+     *
+     * @param reason the reason of the nack, must not be {@code null}
+     * @return a completion stage completed when the message is negative-acknowledgement has completed. If the
+     *         negative acknowledgement fails, the completion stage propagates the failure.
+     */
+    @Experimental("nack support is a SmallRye-only feature")
+    default CompletionStage<Void> nack(Throwable reason) {
+        if (reason == null) {
+            throw new IllegalArgumentException("The reason must not be `null`");
+        }
+        Function<Throwable, CompletionStage<Void>> nack = getNack();
+        if (nack == null) {
+            LOGGER
+                    .warn("A message has been nacked, but no nack function has been provided. The reason was: {}",
```

Comment:
Does `{}` bring in the message? Thought the format was `%s` or the like?

Reply:
Using SLF4J, not JBoss logging.

Reply:
Ah ok, we may want to re-evaluate that in the future